### DISCUSSION TITLE
Error when deploying to Netlify because of 301 redirect with trailing slash

### DISCUSSION THREAD

**Discussion by benoitdemaegdt** [nuxt/content#1410] [Q&A] [answered]
Hey there 👋 

I'm currently building a website with Nuxt3 and Nuxt Content V2, but I'm facing an issue when deploying to Netlify.
Here is the [website](https://monpetitsommet.fr/) and the [github project](https://github.com/benoitdemaegdt/monpetitsommet).

Very similarly to [the official documentation](https://content.nuxtjs.org/guide/displaying/querying#example), I try to fetch a document corresponding to the current browser URL 👇 
```
const { path } = useRoute()

const { data: article } = await useAsyncData(`article-${path}`, () => {
  return queryContent('/blog')
    .where({ _path: path })
    .findOne()
})
```
This works great in dev mode, and in production when navigating using a NuxtLink.
However, when trying to refresh the browser on a page using this kind of query, **the nuxt content query gets a 404 error**.

After some investigations, this is because there is a 301 redirect from `https://website.com/blog/article` to `https://website.com/blog/article/` (note the trailing slash). Because of this trailing slash, the query cannot find any content and there is a 404 error.
<img width="702" alt="image" src="https://user-images.githubusercontent.com/25749578/182479047-6e269c92-8f7a-4f96-a61f-14c99dfb3cbc.png">

I tried to manually remove this trailing slash.
```
const { path } = useRoute()
const { withoutTrailingSlash } = useContent() // simple function removing trailing slash if exists
const { data: article } = await useAsyncData(`article-${path}`, () => {
  return queryContent('/blog')
    .where({ _path: withoutTrailingSlash(path) })
    .findOne()
})
```
This partially works : content is now rendered, but there is an ugly flash happening.

So I have two questions : 
- Should `withoutTrailingSlash` be inside nuxt content ? It seems that it was recently added to the document driven mode (https://github.com/nuxt/content/pull/1293 and https://github.com/nuxt/content/pull/1294), why not for the "standard" mode ?
- Where does this trailing slash 301 redirect comes from ? Is it Nuxt3 behaving this way ? Why ?

Thank you for the amazing work 🙏

**Comment by benoitdemaegdt** [comment, marked as answer]
Problem suddenly disappeared. I am still not sure what was causing it, but 301 redirect to same url with trailing slash is no longer happening.
The solution might be to untick this option in Netlify.
<img width="931" alt="image" src="https://user-images.githubusercontent.com/25749578/183645279-e0d2426a-6b9b-4a6b-9906-7f1d808da117.png">

I also added this middleware to avoid all trailing slash : 
`middleware/noTrailingSlash.global.js`
```js
import { withoutTrailingSlash } from 'ufo'

export default defineNuxtRouteMiddleware((to) => {
  if (to.path.endsWith('/') && to.path !== '/') {
    return navigateTo(withoutTrailingSlash(to.path, true), { redirectCode: 301 })
  }
})
```
Thanks @nathanchase for [the tip](https://github.com/nuxt/framework/discussions/6422).

✌️

**Reply by notflip** [reply]
I had the same issue and it was fixed by unticking the Pretty Url's in Netlify, didn't have to add middleware.
